### PR TITLE
feat(): repository environment variables

### PR DESCRIPTION
**Pull request by Izak88**

### CODE REVIEW

File: src/api/server-routes.ts [hunk 1]
```diff
@@ -402,6 +403,24 @@ export function permissionRoutes(): express.Router {
...
+  const router = express.Router();
+
+  router.post('/add', (req: express.Request, res: express.Response) => {
+    insertEnvironmentVariable(req.body)
+      .then(() => res.status(200).json({ data: true }))
+      .catch(() => res.status(200).json({ data: false }));
+  });
+
+  router.get('/remove/:id', (req: express.Request, res: express.Response) => {
+    removeEnvironmentVariable(req.params.id)
```
Comment (jkuri) [inline src/api/server-routes.ts]: this route is totally unprotected, anyone can delete env variables with accessing this route and break the builds eventually.

File: src/api/server-routes.ts [hunk 2]
```diff
@@ -402,6 +403,24 @@ export function permissionRoutes(): express.Router {
   return router;
 }
 
+export function environmentVariableRoutes(): express.Router {
+  const router = express.Router();
+
+  router.post('/add', (req: express.Request, res: express.Response) => {
+    insertEnvironmentVariable(req.body)
```
Comment (jkuri) [inline src/api/server-routes.ts]: anyone can add env variables via `curl`. please don't be so sloppy.

File: src/api/process.ts [hunk 3]
```diff
@@ -33,48 +33,65 @@ export function startBuildProcess(
...
   return new Observable(observer => {
-    const name = 'abstruse_' + proc.build_id + '_' + proc.job_id;
-    const vars = proc.commands.filter(cmd => cmd.command.startsWith('export'))
-      .map(cmd => cmd.command.replace('export', '-e'))
-      .reduce((acc, curr) => {
-        return acc.concat(curr.split(' '));
-      }, [])
-      .concat(proc.env.reduce((acc, curr) => acc.concat(['-e', curr]), []));
-    proc.commands = proc.commands.filter(cmd => !cmd.command.startsWith('export'));
+    getRepositoryByBuildId(proc.build_id).then(repo => {
```
Comment (jkuri) [inline src/api/process.ts]: why did you add new method for getting repository if `getBuild(proc.build).then(build => build.repository` works just fine?

File: src/api/process.ts [hunk 4]
```diff
@@ -33,48 +33,65 @@ export function startBuildProcess(
...
+      const name = 'abstruse_' + proc.build_id + '_' + proc.job_id;
+      const vars = proc.commands.filter(cmd => cmd.command.startsWith('export'))
+        .map(cmd => cmd.command.replace('export', '-e'))
+        .reduce((acc, curr) => {
+          return acc.concat(curr.split(' '));
+        }, [])
+        .concat(proc.env.reduce((acc, curr) => acc.concat(['-e', curr]), []))
+        .concat(repo.variables.map(v => `${v.name}=${v.value}`)
+          .reduce((acc, curr) => acc.concat(['-e', curr]), []));
```
Comment (jkuri) [inline src/api/process.ts]: če si ti tole stestiral sm js picikl. na 10m se vidi da tole ne more delat?

File: src/api/process.ts (hunk 3 above)
Comment (jkuri) [inline src/api/process.ts]: `repo.variables` is only variable that you need from repository. please add this as parameter to this function and avoid database calls inside `process.ts`.

File: src/api/process.ts (hunk 4 above)
Comment (jkuri) [inline src/api/process.ts]: please fix indent of that reduce two spaces back

**Comment by Izak88** [comment]
PR updated after code review

File: src/api/process.ts [hunk 5]
```diff
@@ -39,7 +40,8 @@ export function startBuildProcess(
       .reduce((acc, curr) => {
         return acc.concat(curr.split(' '));
       }, [])
-      .concat(proc.env.reduce((acc, curr) => acc.concat(['-e', curr]), []));
+      .concat(proc.env.reduce((acc, curr) => acc.concat(['-e', curr]), []))
+      .concat(variables.reduce((acc, curr) => acc.concat(['-e', curr]), []));
```
Comment (jkuri) [inline src/api/process.ts]: 👍

File: src/api/server-routes.ts [hunk 6]
```diff
@@ -276,9 +276,11 @@ export function repositoryRoutes(): express.Router {
   });
 
   router.post('/permission', (req: express.Request, res: express.Response) => {
-    updatePermission(req.body)
-      .then(() => res.status(200).json({ data: true }))
-      .catch(() => res.status(200).json({ data: false }));
+    checkApiRequestAuth(req).then(() => {
+      updatePermission(req.body)
```
Comment (jkuri) [inline src/api/server-routes.ts]: use `return` here and you don't need double catch.

Comment (jkuri) [reply src/api/server-routes.ts]:
```ts
checkApiRequestAuth(req)
  .then(() => updatePermission(req.body))
  .then(() => res.status(200).json({ data: true }))
  .catch(err => res.status(401).json({ data: 'Not Authorized' }));
```

File: src/api/process-manager.ts [hunk 7]
```diff
@@ -170,8 +170,9 @@ export function startBuild(data: any): Promise<any> {
 
 export function startJob(proc: JobProcess): Promise<void> {
   return Promise.resolve()
-    .then(() => {
-      startBuildProcess(proc, 'abstruse')
+    .then(() => getRepositoryByBuildId(proc.build_id))
+    .then(repo => {
+      startBuildProcess(proc, 'abstruse', repo.variables.map(v => `${v.name}=${v.value}`))
```
Comment (jkuri) [inline src/api/process-manager.ts]: this actually works but please move `repo.variables.map(v => `${v.name}=${v.value}`)` to variable and pass that to function.
